c2corg_api/views/document.py

```diff
@@ -311,40 +311,55 @@ def _delete(
         if document.type == IMAGE_TYPE:
             filenames = DBSession.query(ArchiveImage.filename). \
                 filter(ArchiveImage.document_id == id).all()
-            # FIXME remove debug when var is used
-            print('DEBUG FILENAMES', filenames)
 
         # Note: if an error occurs while deleting, SqlAlchemy will
         # automatically cancel all changes.
 
+        # Remove document from the cache system
+        DBSession.query(CacheVersion). \
+            filter(CacheVersion.document_id == id).delete()
+
+        # Remove associations and update cache of formerly associated docs
+        doc_associations.remove_associations(document)
+        update_cache_version(document)
```

Comment:
I guess you have to call `update_cache_version` before removing the associations. Otherwise the cache versions of the associations cannot be updated.